kore/test/Test/Kore/Equation/Application.hs

```diff
@@ -421,6 +588,45 @@ axiom_
     -> Equation'
 axiom_ left right = axiom left right (makeTruePredicate sortR)
 
+functionAxiomNEW
```

Comment:
Let's call this `functionAxiomUnification`, to emphasize that it constructs a function axiom using unification for the arguments.